src/dodal/devices/diamond_filter.py

```diff
@@ -0,0 +1,46 @@
+from enum import Enum
+from typing import Generic, TypeVar
+
+from ophyd_async.core import StandardReadable
+from ophyd_async.epics.motor import Motor
+from ophyd_async.epics.signal import epics_signal_rw
+
+
+class Filters(str, Enum):
+    pass
+
+
+class I03Filters(Filters):
+    EMPTY = "Empty"
+    TWO_HUNDRED = "200um"
+    ONE_HUNDRED = "100um"
+
+
+class I04Filters(Filters):
+    EMPTY = "Empty"
+    TWO_HUNDRED = "200um"
+    FIFTY = "50um"
+    OUT = "Out"
+
+
+T = TypeVar("T", bound=Filters)
+
+
+class DiamondFilter(StandardReadable, Generic[T]):
```

Comment:
@coretl or @callumforrester. Do we like this pattern for the different enums on each beamline or do you think better to just drop it and leave it as a `str`?

Reply:
Depends, do you in code write `filter.thickness.set(I03Filters.FIFTY)` anywhere? If so then it's useful to check that the specified filter value exists at connect too. If not then drop to string.

Reply:
I'm not sure I understand. We don't write `filter.thickness.set(I03Filters.FIFTY)` anywhere because there is no 50um filter on I03. The filters are different on each beamline but otherwise the hardware is the same. What I'm trying to do is allow people to write:
```python
i03_filter = i03.filter()
i04_filter = i04.filter()

i03_filter.thickness.set(I03Filters.TWO_HUNDRED)  # All good
i03_filter.thickness.set(I04Filters.FIFTY) # Type system gives an error here, there is no 50 on i03
```
My concern with `str` was that you can then do:
```python 
i03_filter.thickness.set("200 um")  # All good
i03_filter.thickness.set("50 um") # No warning this is not valid until runtime
```
I wanted to make sure this pattern wasn't overly complicating things or doing something unexpected for `ophyd_async` though

Reply:
The procedural case will remain compatible with this, but making the declarative approach work for this would be overkill. I'd say it makes sense for the usecase you mention.